drivers/pinmux/pinmux_stm32.c

```diff
@@ -187,78 +187,91 @@ int stm32_dt_pinctrl_remap(const struct soc_gpio_pinctrl *pinctrl,
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(can1), okay)
 	case DT_REG_ADDR(DT_NODELABEL(can1)):
 		if (remap == REMAP_1) {
-			/* PB8/PB9 */
-			LL_GPIO_AF_RemapPartial2_CAN1();
+			/* PB8/PB9 (CAN_REMAP = 0b10) */
+			MODIFY_REG(AFIO->MAPR, AFIO_MAPR_CAN_REMAP,
```

Comment:
Using CMSIS API to modify this register may cause instability:
as per reference manual:
![image](https://user-images.githubusercontent.com/50730894/134675809-24e3b2a2-2f6c-4f53-9870-62bd15c8680c.png)

So when modifying the register, it starts by reading the register, with undetermined value for SWJ_CFG, 
this undetermined value is then written back to the register. Leading to undetermined behavior of SWJ

This is for me the reason why LL API, manipulates SWJ_CFG sytematically for AFIO_MAPR register and set an invalid SWJ_CFG value to have no effect.
So I think we should keep the original LL API.

Reply:
For remapping SPI1, we will have:
```
		/* enable remap */
#if defined(GPIO_STM32_SWJ_ENABLE)
	/* reset state */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			(AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG_RESET));
#elif defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
	/* released PB4 */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			(AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG_NOJNTRST));
#elif defined(CONFIG_GPIO_STM32_SWJ_NOJTAG)
	/* released PB4 PB3 PA15 */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			(AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG_JTAGDISABLE));
#elif defined(CONFIG_GPIO_STM32_SWJ_DISABLE)
	/* released PB4 PB3 PA13 PA14 PA15 */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			(AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG_DISABLE));
#endif
```
For disabling remap of SPI1, we will have:
```
		/* disable remap */
#if defined(GPIO_STM32_SWJ_ENABLE)
	/* reset state */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			AFIO_MAPR_SWJ_CFG_RESET);
#elif defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
	/* released PB4 */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			AFIO_MAPR_SWJ_CFG_NOJNTRST);
#elif defined(CONFIG_GPIO_STM32_SWJ_NOJTAG)
	/* released PB4 PB3 PA15 */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			AFIO_MAPR_SWJ_CFG_JTAGDISABLE);
#elif defined(CONFIG_GPIO_STM32_SWJ_DISABLE)
	/* released PB4 PB3 PA13 PA14 PA15 */
	MODIFY_REG(AFIO->MAPR, (AFIO_MAPR_SPI1_REMAP | AFIO_MAPR_SWJ_CFG),
			AFIO_MAPR_SWJ_CFG_DISABLE);
#endif	
```